openedx/core/djangoapps/catalog/utils.py

```diff
@@ -136,6 +138,9 @@ def get_programs_with_type(types=None):
             if types and program['type'] not in types:
                 continue
 
+            if program.get('hidden', False) and not include_hidden:
```

Comment:
@rlucioni I had another PR (https://github.com/edx/edx-platform/pull/15226) where hidden programs were not being cached, and the API was responsible for the filtering. However, @clintonb mentioned that we need to cache hidden programs.